### PR TITLE
Kf 2267 feat jupyter pytorch cuda full rock

### PR DESCRIPTION
**Pull request by i-chvets**
Summary of changes:
    - Added security requirements and updated non-root user.

Based on #3

### CODE REVIEW

File: jupyter-pytorch-cuda-full/rockcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+      PATH: /opt/conda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+      NVIDIA_VISIBLE_DEVICES: all
+      NVIDIA_DRIVER_CAPABILITIES: compute,utility
+      LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
+    command: /opt/conda/bin/jupyter lab --notebook-dir="/home/jovyan" --ip=0.0.0.0 --no-browser --allow-root --port=8888 --ServerApp.token="" --ServerApp.password="" --ServerApp.allow_origin="*" --ServerApp.base_url="/" --ServerApp.authenticate_prometheus=False
+platforms:
+  amd64:
+
+parts:
+  overlay-pkgs:
```
Comment (misohu) [inline jupyter-pytorch-cuda-full/rockcraft.yaml:37]: Is this part even useful ? alll these packages are installed but nothing happens in this part.

File: jupyter-pytorch-cuda-full/rockcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+  kubectl:
+    plugin: nil
+    stage-snaps:
+      - kubectl/1.21/stable
+    organize:
+      kubectl: bin/kubectl
+    stage:
+      - bin/kubectl
+
+  nodejs:
```
Comment (misohu) [inline jupyter-pytorch-cuda-full/rockcraft.yaml:68]: Is this part even useful ? alll these packages are installed but nothing happens in this part.

Comment (i-chvets) [reply jupyter-pytorch-cuda-full/rockcraft.yaml:68]: But `organize` and `stage` should have effect, right?

Comment (i-chvets) [reply jupyter-pytorch-cuda-full/rockcraft.yaml:68]: > Is this part even useful ? alll these packages are installed but nothing happens in this part.

But organize and stage should have effect, right?

**Comment by i-chvets** [comment]
Tested:
```
$ docker run jupyter-pytorch-cuda-full_v1.7.0_22.04_1_amd64:rock exec pebble plan
2023-05-17T17:38:24.552Z [pebble] Started daemon.
2023-05-17T17:38:24.568Z [pebble] POST /v1/exec 14.587426ms 202
2023-05-17T17:38:24.584Z [pebble] GET /v1/tasks/1/websocket/control 14.924612ms 200
2023-05-17T17:38:24.585Z [pebble] GET /v1/tasks/1/websocket/stdio 206.043µs 200
2023-05-17T17:38:24.586Z [pebble] GET /v1/tasks/1/websocket/stderr 111.955µs 200
2023-05-17T17:38:24.591Z [pebble] GET /v1/plan?format=yaml 958.562µs 200
services:
    jupyter:
        summary: jupyter-pytorch-cuda-full service
        startup: enabled
        override: replace
        command: /opt/conda/bin/jupyter lab --notebook-dir="/home/jovyan" --ip=0.0.0.0 --no-browser --allow-root --port=8888 --ServerApp.token="" --ServerApp.password="" --ServerApp.allow_origin="*" --ServerApp.base_url="/" --ServerApp.authenticate_prometheus=False
        environment:
            HOME: /home/jovyan
            LANG: en_US.UTF-8
            LANGUAGE: en_US.UTF-8
            LC_ALL: en_US.UTF-8
            LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
            NB_PREFIX: /
            NB_UID: "1001"
            NB_USER: jovyan
            NVIDIA_DRIVER_CAPABILITIES: compute,utility
            NVIDIA_VISIBLE_DEVICES: all
            PATH: /opt/conda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
            SHELL: /bin/bash
        user: jovyan
2023-05-17T17:38:24.620Z [pebble] GET /v1/changes/1/wait 33.614559ms 200

$ docker run jupyter-pytorch-cuda-full_v1.7.0_22.04_1_amd64:rock exec /opt/conda/bin/jupyter lab --notebook-dir="/home/jovyan" --ip=0.0.0.0 --no-browser --allow-root --port=8888 --ServerApp.token="" --ServerApp.password="" --ServerApp.allow_origin="*" --ServerApp.base_url="/" --ServerApp.authenticate_prometheus=False
2023-05-17T17:38:33.190Z [pebble] Started daemon.
2023-05-17T17:38:33.207Z [pebble] POST /v1/exec 16.048378ms 202
2023-05-17T17:38:33.223Z [pebble] GET /v1/tasks/1/websocket/control 13.99738ms 200
2023-05-17T17:38:33.224Z [pebble] GET /v1/tasks/1/websocket/stdio 104.792µs 200
2023-05-17T17:38:33.224Z [pebble] GET /v1/tasks/1/websocket/stderr 80.659µs 200
[I 2023-05-17 17:38:35.002 ServerApp] jupyter_server_mathjax | extension was successfully linked.
[I 2023-05-17 17:38:35.005 ServerApp] jupyterlab | extension was successfully linked.
[I 2023-05-17 17:38:35.005 ServerApp] jupyterlab_git | extension was successfully linked.
[I 2023-05-17 17:38:35.007 ServerApp] nbclassic | extension was successfully linked.
[I 2023-05-17 17:38:35.007 ServerApp] nbdime | extension was successfully linked.
[I 2023-05-17 17:38:35.008 ServerApp] Writing Jupyter server cookie secret to /root/.local/share/jupyter/runtime/jupyter_cookie_secret
[I 2023-05-17 17:38:35.703 ServerApp] notebook_shim | extension was successfully linked.
[W 2023-05-17 17:38:35.753 ServerApp] All authentication is disabled.  Anyone who can connect to this server will be able to run code.
[I 2023-05-17 17:38:35.756 ServerApp] notebook_shim | extension was successfully loaded.
[I 2023-05-17 17:38:35.756 ServerApp] jupyter_server_mathjax | extension was successfully loaded.
[I 2023-05-17 17:38:35.757 LabApp] JupyterLab extension loaded from /opt/conda/lib/python3.8/site-packages/jupyterlab
[I 2023-05-17 17:38:35.757 LabApp] JupyterLab application directory is /opt/conda/share/jupyter/lab
[I 2023-05-17 17:38:35.758 ServerApp] jupyterlab | extension was successfully loaded.
[I 2023-05-17 17:38:35.760 ServerApp] jupyterlab_git | extension was successfully loaded.
[I 2023-05-17 17:38:35.767 ServerApp] nbclassic | extension was successfully loaded.
[I 2023-05-17 17:38:35.896 ServerApp] nbdime | extension was successfully loaded.
[I 2023-05-17 17:38:35.897 ServerApp] Serving notebooks from local directory: /home/jovyan
[I 2023-05-17 17:38:35.897 ServerApp] Jupyter Server 1.24.0 is running at:
[I 2023-05-17 17:38:35.897 ServerApp] http://c9dc8168fd86:8888/lab
[I 2023-05-17 17:38:35.897 ServerApp]  or http://127.0.0.1:8888/lab
[I 2023-05-17 17:38:35.897 ServerApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
```

File: jupyter-pytorch-cuda-full/rockcraft.yaml (hunk 1 above)
Comment (i-chvets) [reply jupyter-pytorch-cuda-full/rockcraft.yaml:53]: Looks like those packages will be available for other parts:
https://canonical-craft-parts.readthedocs-hosted.com/en/latest/explanation/overlays.html#overlay-parameters

**Review by misohu — APPROVED** [review]

**Pull request merged**